src/webgpu/api/validation/createBindGroupLayout.spec.ts

```diff
@@ -88,6 +94,96 @@ g.test('mininum_buffer_binding_size')
     }, !success);
   });
 
+g.test('view_dimension')
+  .params(
+    params()
+      .combine(poptions('type', kBindingTypes))
+      .combine(poptions('viewDimension', [undefined, '2d', 'cube'] as const))
+  )
+  .fn(async t => {
+    const { type, viewDimension } = t.params;
+
+    let success = false;
+    if (
+      viewDimension === undefined ||
+      (viewDimension in kTextureViewDimensionInfo && type in kTextureBindingTypeInfo)
+    ) {
+      success = true;
+    }
+
+    t.expectValidationError(() => {
+      t.device.createBindGroupLayout({
+        entries: [{ binding: 0, visibility: GPUShaderStage.COMPUTE, type, viewDimension }],
+      });
+    }, !success);
+  });
+
+g.test('texture_component_type')
+  .params(
+    params()
+      .combine(poptions('type', kBindingTypes))
+      .combine(poptions('textureComponentType', [undefined, 'float', 'uint'] as const))
+  )
+  .fn(async t => {
+    const { type, textureComponentType } = t.params;
+
+    let success = false;
+    if (
+      textureComponentType === undefined ||
+      (textureComponentType in kTextureComponentTypeInfo && type === 'sampled-texture')
+    ) {
+      success = true;
+    }
+
+    t.expectValidationError(() => {
+      t.device.createBindGroupLayout({
+        entries: [{ binding: 0, visibility: GPUShaderStage.COMPUTE, type, textureComponentType }],
+      });
+    }, !success);
+  });
+
+g.test('multisampled')
+  .params(params().combine(poptions('type', kBindingTypes)).combine(pbool('multisampled')))
+  .fn(async t => {
+    const { type, multisampled } = t.params;
+
+    let success = false;
+    if (multisampled === false || type === 'sampled-texture') {
+      success = true;
+    }
+
+    t.expectValidationError(() => {
+      t.device.createBindGroupLayout({
+        entries: [{ binding: 0, visibility: GPUShaderStage.COMPUTE, type, multisampled }],
+      });
+    }, !success);
+  });
+
+g.test('storage_texture_format')
+  .params(
+    params()
+      .combine(poptions('type', kBindingTypes))
+      .combine(poptions('storageTextureFormat', [undefined, 'rgba8unorm'] as const))
```

Comment:
I guess there should be validation that storageTextureFormat can only be a format which supports storage usage. So we should test all possible texture formats here and use `kTextureFormatInfo[storageTextureFormat].storage`.

Reply:
I don't think the spec requires that we validate that the `storage-texture-format` is a format that supports storage usage. Maybe we should add it.

Reply:
Yes. I realized this too yesterday. a new PR is just submitted at https://github.com/gpuweb/gpuweb/pull/925. Did we specify which formats can support storage usage in spec? I can't find the statement in spec.

Reply:
@kainino0x , will test every format with a follow-up CL, together with refactoring.

Reply:
Yeah, it's not in the spec. We don't have a format table in the spec yet either; I've been using the one in the CTS as the source of truth, but we need to port it over to the spec.